olive/engine/engine.py

```diff
@@ -910,6 +919,17 @@ def _run_pass(
 
         run_start_time = datetime.now().timestamp()
         try:
+            from olive.passes.onnx.perf_tuning import OrtPerfTuning
+
+            # For perf-tuning, we need the model evaluation happens in target even if it is a pass.
```

Comment:
I agree this is an issue. I brought this up even during the initial design of Olive v2 that there are some passes like perf tuning which require to be run on the target. 

It is also similar for other passes like Inc quantization where they do latency evaluation. But that is a more complicated scenario than this.

Reply:
Yes, from the time being, it is a hack way to workaround current bug. we need consider more elegant way to fix this bug systematically.